src/extension.ts

```diff
@@ -80,6 +78,17 @@ export async function activate(context: vscode.ExtensionContext): Promise<Api |
             });
 
         context.subscriptions.push(...commands.registerToolchainCommands(toolchain));
+
+        if (!toolchain) {
+            showToolchainError();
+            return;
+        }
+
+        const workspaceContext = await WorkspaceContext.create(outputChannel, toolchain);
+        context.subscriptions.push(...commands.register(workspaceContext));
+        context.subscriptions.push(workspaceContext);
+        context.subscriptions.push(registerDebugger(workspaceContext));
+
         context.subscriptions.push(
```

Comment:
The onDidChangeConfiguration() handler needs to be registered alongside the toolchain commands. This dialog should appear even if there is no WorkspaceContext so that the Select `Toolchain...` command can tell users to reload the extension host.

Reply:
Done